packages/react-components/react-menu/src/components/MenuTrigger/useMenuTrigger.ts

```diff
@@ -128,19 +122,22 @@ export const useMenuTrigger_unstable = (props: MenuTriggerProps): MenuTriggerSta
 
   return {
     isSubmenu,
-    children: applyTriggerPropsToChildren<MenuTriggerChildProps>(children, {
-      'aria-haspopup': 'menu',
-      'aria-expanded': !open && !isSubmenu ? undefined : open,
-      id: triggerId,
-      ...child?.props,
-      ref: useMergedRefs(triggerRef, child?.ref),
-      onClick: useMergedEventCallbacks(child?.props?.onClick, onClick),
-      onMouseEnter: useMergedEventCallbacks(child?.props?.onMouseEnter, onMouseEnter),
-      onMouseLeave: useMergedEventCallbacks(child?.props?.onMouseLeave, onMouseLeave),
-      onKeyDown: useMergedEventCallbacks(child?.props?.onKeyDown, onKeyDown),
-      onContextMenu: useMergedEventCallbacks(child?.props?.onContextMenu, onContextMenu),
-      onMouseMove: useMergedEventCallbacks(child?.props?.onMouseMove, onMouseMove),
-    }),
+    children: applyTriggerPropsToChildren<MenuTriggerChildProps>(
+      children,
+      useARIAButtonProps(child?.type === 'button' || child?.type === 'a' ? child.type : 'div', {
```

Comment:
would this work if someone used the fluent `Button`? the `type` should be different so it would spread all the props of a `div` button

Reply:
No, it would consider fluent `Button` as a `div` and provide all handlers all over again, but that wouldn't cause any issues, as multiple instances of `useARIAButton` should not conflict with each other.

The ideal solution however would be to have some sort of annotation that indicates a certain component type already uses `useARIAButton` internally, and therefore does not need any extra handling.

Reply:
> The ideal solution however would be to have some sort of annotation that indicates a certain component type already uses useARIAButton internally, and therefore does not need any extra handling.

But this is definitely out of scope of this PR